src/set.js

```diff
@@ -0,0 +1,27 @@
+var _curry2 = require('./internal/_curry2');
+var _set = require('./internal/_set');
+
+
+/**
+ * Creates a new object with the values from `b` merged with the
+ * values in `a`. Only own enumerable properties are considered. This function
+ * recursively handles nested objects, but not arrays, with own enumerable
+ * properties. All non-primitive properties are copied by reference if
+ * possible. This function * will *not* mutate passed-in objects.
+ *
+ * @func
+ * @memberOf R
+ * @category Object
+ * @sig {k: v} -> {k: v} -> {k: v}
+ * @param {Object} a object with higher precedence in output
+ * @param {Object} b source object
+ * @return {Object} The merged object.
+ * @example
+ *
+ *      R.set({age: 40, {name: 'fred', age: 10});
+ *      //=> {'name': 'fred', 'age': 40}
+ *
+ *      R.set({parent: {age: 40}}, {name: 'fred', age: 10, parent: {age: 39}});
+ *      //=> {'name': 'fred', 'age': 10, {parent: {age: 40}}}
+ */
+module.exports = _curry2(_set);
```

Comment:
I don't see any reason for the internal/external distinction here.  We need to do that if there are internal consumers of the uncurried/undispatched version.  (Perhaps.  Even then, I would often prefer usually to dogfood our stuff.  But here the only use of the internal function is the recursive use.  That will be captured by the named function expression.